tests/test_preload_script.py

```diff
@@ -95,3 +95,290 @@ async def test_addPreloadScript_logging(websocket, context_id):
     # Assert navigation is finished.
     result = await read_JSON_message(websocket)
     assert result == {"id": command_id, "result": ANY_DICT}
+
+
+# TODO: make offline.
+@pytest.mark.parametrize("nested_iframe", [
+    'about:blank',
+    'https://example.com/',
+])
+@pytest.mark.asyncio
+async def test_addPreloadScript_loadedInNewIframes(websocket, context_id,
+                                                   nested_iframe):
+    await subscribe(websocket, "log.entryAdded")
+
+    await execute_command(
+        websocket, {
+            "method": "script.addPreloadScript",
+            "params": {
+                "expression": "() => console.log('my preload script')",
+                "context": context_id,
```

Comment:
Consider parametrize `globally` / `for a given context` as well.
```suggestion
@pytest.mark.parametrize("nested_iframe", [
    'about:blank',
    'https://example.com/',
])
@pytest.mark.parametrize("specific_context", [True, False])
@pytest.mark.asyncio
async def test_addPreloadScript_loadedInNewIframes(websocket, context_id,
                                                   nested_iframe):
    await subscribe(websocket, "log.entryAdded")

    await execute_command(
        websocket, {
            "method": "script.addPreloadScript",
            "params": (specific_context ? {"context": context_id} : {}) | {
                "expression": "() => console.log('my preload script')",
```